src/engine/engine.cpp

```diff
@@ -22,27 +18,28 @@
 
 namespace
 {
+
+auto GetWatchdogDataFacade()
```

Comment:
I agree a singleton is an ugly solution here, however it has at least the upside that we only have one update thread per process.

Thinking ahead a little bit, I'm not sure how we could extend this to multiple shared-memory regions (to have multiple datasets in shared memory at the same time).

Maybe we could circumvent the need for a central control region with message based IPC for updates and a control file for persistent storage. I'm going to open an issue about this, maybe we can come up with more ideas. For now this is fine. 👍